### PR TITLE
Revenue cli

### PR DESCRIPTION
**Pull request by JssDWt**
This PR adds a cli for revenue data.
There are 2 commands:
- `export-forwards`
- `revenue`

```
NAME:
   lspd_revenue_cli - get revenue data from lspd

USAGE:
   lspd_revenue_cli [global options] command [command options] [arguments...]

VERSION:
   v24.01.14-36-gf093584 commit=f093584bab127fe905831bb28c1b875230a3b9a8

COMMANDS:
   export-forwards  Export forwards with a given peer correlated to an api key for a given time period.
   revenue          Get a revenue report.
   help, h          Shows a list of commands or help for one command

GLOBAL OPTIONS:
   --database-url value  postgres database url for lspd
   --help, -h            show help
   --version, -v         print the version
```

This PR depends on https://github.com/breez/lspd/pull/189
The data about fees used for channel opens will only be reliable if the above PR was running throughout the entire time period the revenue data is fetched for.

### CODE REVIEW

File: cmd/lspd_revenue_cli/revenue.go [hunk 1]
```diff
@@ -0,0 +1,648 @@
...
+		if forward.SendToken != nil {
+			sendToken, ok := currentNode.Tokens[*forward.SendToken]
+			if !ok {
+				sendToken = &TokenRevenue{
+					Token: *forward.SendToken,
+				}
+				currentNode.Tokens[*forward.SendToken] = sendToken
+			}
+
+			feesMsat := (forward.AmtMsatIn - forward.AmtMsatOut) / 2
```
Comment (hydra-yse) [inline cmd/lspd_revenue_cli/revenue.go:573]: Just to understand, why do we split the fees in two?

**Review by hydra-yse — APPROVED** [review]
Looking good Jesse! Just those `importedIndex` changes, and it should be ready.

File: cmd/lspd_revenue_cli/revenue.go [hunk 2]
```diff
@@ -0,0 +1,648 @@
...
+		}
+
+		if importedIndex >= len(importedForwards) {
+			break
+		}
+
+		importedForward := importedForwards[importedIndex]
+		if importedForward.direction != "send" {
+			continue
+		}
```
[the remark this answers is not in the document]
Comment (JssDWt) [reply cmd/lspd_revenue_cli/revenue.go]: Thanks, done

File: cmd/lspd_revenue_cli/revenue.go [hunk 3]
```diff
@@ -0,0 +1,648 @@
...
+
+		if importedIndex >= len(importedForwards) {
+			break
+		}
+
+		importedForward := importedForwards[importedIndex]
+		if importedForward.direction != "receive" {
+			continue
+		}
```
[the remark this answers is not in the document]
Comment (JssDWt) [reply cmd/lspd_revenue_cli/revenue.go]: Done

**Review by yaslama — APPROVED** [review]

**Pull request merged**